bindsnet/network/topology.py

```diff
@@ -157,21 +157,24 @@ def __init__(
         """
         super().__init__(source, target, nu, weight_decay, **kwargs)
 
-        self.w = kwargs.get("w", None)
-        if self.w is None:
+        w = kwargs.get("w", None)
```

Comment:
Is there a redone to this use of internal self.w and temporal w?

Reply:
I wanted to set it as a `Parameter` and you can't pass `None` into that, only `torch.Tensor`s. Thus for the initial logic of creating `self.w` this is done without assigning to `self.w` and instead creating a temporary `w` that doesn't have the same constraints.